server/chat-plugins/poll.ts

```diff
@@ -572,12 +662,13 @@ export const pages: PageTable = {
 				`(${this.tr`delete`})</button>`
 			);
 			buf += `<hr />`;
-			buf += `${button}<br />${poll.generateResults()}`;
+			buf += `${button}<br />${new Poll(room, poll).generateResults()}`;
```

Comment:
Errrr... this seems like it would cause bugs.

You should maybe do a `static generateResults` that can generate them from a `PollData`?

Reply:
For convenience, maybe also let `Poll implements PollData`?

Reply:
This is getting kind of involved, which I'm sorry about. I unfortunately don't think this sort of thing is mergeable, but if you're getting overwhelmed, just say so and I can work on it myself.

Reply:
No worries.